rezervo/providers/sats/provider.py

```diff
@@ -0,0 +1,311 @@
+                        (
+                            day
+                            for day in (sats_schedule.events or [])
+                            if day.date
+                            == (from_date + datetime.timedelta(days=i)).strftime(
+                                "%Y-%m-%d"
+                            )
+                        ),
+                        SatsDaySchedule(
+                            date=(from_date + datetime.timedelta(days=i)).strftime(
+                                "%Y-%m-%d"
+                            ),
+                            events=[],
+                        ),
+                    )
+                    for i in range(days)
+                ]
+            )
+        )
+
+    def _rezervo_schedule_from_sats_schedule(
+        self, schedule: SatsWeekSchedule
+    ) -> RezervoSchedule:
+        return RezervoSchedule(
+            days=[
+                RezervoDay(
+                    day_name=WEEKDAYS[
+                        datetime.datetime.fromisoformat(day_schedule.date).weekday()
+                    ],
+                    date=datetime.datetime.fromisoformat(day_schedule.date).isoformat(),
+                    classes=[
+                        self.rezervo_class_from_sats_class(c)
+                        for c in day_schedule.events
+                    ],
+                )
+                for day_schedule in (schedule.events or [])
+            ]
+        )
+
+    def rezervo_class_from_sats_class(
+        self,
+        sats_class: SatsClass,
+    ) -> RezervoClass:
+        category = determine_activity_category(sats_class.metadata.name)
+        start_time = datetime.datetime.fromisoformat(sats_class.metadata.startsAt)
+
+        location_id_match = re.search(r"(\d+)p", sats_class.id)
+        if not location_id_match:
+            raise Exception("Could not retrieve location id from Sats class")
+
+        return RezervoClass(
+            id=sats_class.id,
+            start_time=start_time,
+            end_time=start_time
+            + datetime.timedelta(minutes=sats_class.metadata.duration),
+            location=RezervoLocation(
+                id=self.location_from_provider_location_identifier(
+                    int(location_id_match.group(1))
+                ),
+                studio=sats_class.metadata.clubName,
+                room="",  # Sats does not specify rooms
+            ),
+            is_bookable=start_time
+            - datetime.timedelta(days=7)  # https://www.sats.no/legal/bookingregler
+            < datetime.datetime.now().astimezone()
+            < start_time - datetime.timedelta(minutes=10),
+            is_cancelled=False,  # Sats seemingly does not expose a class is canceled
+            total_slots=None,
+            available_slots=None,
+            waiting_list_count=sats_class.waitingListCount,
+            activity=RezervoActivity(
+                id=sats_class.metadata.name,  # Sats does not provide an activity ids
```

Comment:
This might work, but I suspect that the recurrent id will mess up the cli command in cron jobs (because of spaces and other weird chars). What about using a non-crypto hash to create shorter but still (sufficiently) unique ids? [xxhash](https://pypi.org/project/xxhash/) looks promising

Reply:
Implemented in https://github.com/mathiazom/rezervo/pull/43/commits/4ab071ef60185945e9c1fe7cd396e4174c4f9232